### PR TITLE
Upgrade material-ui to v5

### PR DESCRIPTION
**Pull request by conundrumgirl**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,8 +3,9 @@
   "version": "0.0.1",
   "private": true,
   "dependencies": {
-    "@material-ui/core": "^4.12.3",
-    "@material-ui/icons": "^4.9.1",
+    "@mui/icons-material": "^5.11.0",
```
Comment (conundrumgirl) [inline package.json:6]: package names have changed

File: src/App.tsx [hunk 2]
```diff
@@ -1,56 +1,71 @@
-import React from 'react'
-import './App.scss'
-import AppInitializer from './AppInitializer'
-import { SynapseContextConsumer, SynapseContextType } from 'synapse-react-client/dist/utils/SynapseContext'
+
+import { createTheme, StyledEngineProvider, Theme, ThemeProvider } from '@mui/material/styles';
```
Comment (conundrumgirl) [inline src/App.tsx:2]: my formatter rearranges imports as vendor->local alphabetically

File: src/App.tsx [hunk 3]
```diff
@@ -1,56 +1,71 @@
...
-import { AccountSettings } from 'components/AccountSettings'
-import { AppContextConsumer } from 'AppContext'
+} from 'react-router-dom';
+import { SynapseComponents } from 'synapse-react-client';
+import { signOut } from 'synapse-react-client/dist/utils/SynapseClient';
+import { SynapseContextConsumer, SynapseContextType } from 'synapse-react-client/dist/utils/SynapseContext';
+import './App.scss';
+import AppInitializer from './AppInitializer';
+import LoginPage from './LoginPage';
+import generalTheme from './style/theme';
```
Comment (conundrumgirl) [inline src/App.tsx:28]: this file now empty but will potentially host common component styles

File: src/App.tsx [hunk 4]
```diff
@@ -1,56 +1,71 @@
...
+
+
+declare module '@mui/styles/defaultTheme' {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface (remove this line if you don't have the rule enabled)
+  interface DefaultTheme extends Theme { }
+}
+
+// theme is a merge of a general theme and particular color pallettesfor the source app
+const theme = createTheme(deepmerge(getSourceAppTheme(), generalTheme));
```
Comment (conundrumgirl) [inline src/App.tsx:37]: new app theme

File: src/App.tsx [hunk 5]
```diff
@@ -1,56 +1,71 @@
-import React from 'react'
-import './App.scss'
-import AppInitializer from './AppInitializer'
-import { SynapseContextConsumer, SynapseContextType } from 'synapse-react-client/dist/utils/SynapseContext'
+
+import { createTheme, StyledEngineProvider, Theme, ThemeProvider } from '@mui/material/styles';
+import "@mui/styles";
+import { deepmerge } from '@mui/utils';
```
Comment (conundrumgirl) [inline src/App.tsx:4]: new: to merge theme files to create a theme

File: src/App.tsx [hunk 6]
```diff
@@ -1,56 +1,71 @@
...
+// theme is a merge of a general theme and particular color pallettesfor the source app
+const theme = createTheme(deepmerge(getSourceAppTheme(), generalTheme));
 
 const App: React.FC = () => {
   return (
     <div className="App">
-      <>
-        <Router>
+      <StyledEngineProvider injectFirst>
+        <ThemeProvider theme={theme}>
```
Comment (conundrumgirl) [inline src/App.tsx:43]: [injection order](https://mui.com/material-ui/migration/v5-style-changes/#%E2%9C%85-adjust-css-injection-order) and ThemeProvider

File: src/App.tsx [hunk 7]
```diff
@@ -1,56 +1,71 @@
...
                         } else {
                           return <AppContextConsumer>
-                          {appContext => (
-                            <>
-                              {appContext?.redirectURL && window.location.replace(appContext?.redirectURL)}
-                            </>
-                          )}
-                        </AppContextConsumer>
+                            {appContext => (
+                              <>
```
Comment (conundrumgirl) [inline src/App.tsx:58]: formatting

File: src/AppInitializer.tsx [hunk 8]
```diff
@@ -1,17 +1,20 @@
...
 import { SynapseClient } from 'synapse-react-client'
 import { SynapseContextProvider } from 'synapse-react-client/dist/utils/SynapseContext'
 import { UserProfile } from 'synapse-react-client/dist/utils/synapseTypes'
-import useAnalytics from './useAnalytics'
 import { getSearchParam } from 'URLUtils'
-import { ThemeOptions } from '@mui/material'
-import { getSourceAppTheme } from 'components/SourceApp'
-import { Redirect, useLocation } from 'react-router-dom'
-import { AppContextProvider } from 'AppContext'
+import theme from './style/theme'
```
Comment (conundrumgirl) [inline src/AppInitializer.tsx:16]: new theme file and import formatting

File: src/AppInitializer.tsx [hunk 9]
```diff
@@ -96,7 +99,7 @@ function AppInitializer(props: { children?: React.ReactNode }) {
   const [isFramed, setIsFramed] = useState(false)
   const [appId, setAppId] = useState<string>()
   const [redirectURL, setRedirectURL] = useState<string>()
-  const [themeOptions, setThemeOptions] = useState<ThemeOptions>()
+  const [themeOptions, setThemeOptions] = useState<DeprecatedThemeOptions>()
```
Comment (conundrumgirl) [inline src/AppInitializer.tsx:102]: ThemeOptions have has changed in v5. They provide Deprecated for compatibility. Might be able to fix it in more elegant way

File: src/AppInitializer.tsx [hunk 10]
```diff
@@ -137,7 +140,7 @@ function AppInitializer(props: { children?: React.ReactNode }) {
 
   useEffect(() => {
     if (appId) {
-      setThemeOptions(getSourceAppTheme())
+      setThemeOptions(deepmerge(theme, getSourceAppTheme()))
```
Comment (conundrumgirl) [inline src/AppInitializer.tsx:143]: theme: merge of app theme files (control styles if needed, and source pallette)

File: src/LoginPage.tsx [hunk 11]
```diff
@@ -1,15 +1,17 @@
 import * as React from 'react'
-import { withRouter, RouteComponentProps } from 'react-router-dom'
```
Comment (conundrumgirl) [inline src/LoginPage.tsx:2]: everything here is formatting

File: src/components/ProfilePage.tsx [hunk 12]
```diff
@@ -1,20 +1,19 @@
...
 import EditIcon from 'assets/RedEditPencil.svg'
 import VerifiedBorder from 'assets/VerifiedProfilePic.svg'
-import { SynapseConstants, Typography } from 'synapse-react-client'
-import Skeleton from '@mui/material/Skeleton'
-import { SkeletonTable } from 'synapse-react-client/dist/assets/skeletons/SkeletonTable'
+import React, { useEffect, useState } from 'react'
+import { Button, Col, Container, FormControl, FormGroup, FormLabel, Modal, Row } from 'react-bootstrap'
 import Cropper from "react-easy-crop"
 import { Area } from 'react-easy-crop/types'
-import Slider from "@material-ui/core/Slider"
```
Comment (conundrumgirl) [inline src/components/ProfilePage.tsx:17]: formatting. The reason for update is the change in @mui package name in v5

File: src/components/RegisterAccount1.tsx [hunk 13]
```diff
@@ -58,9 +58,14 @@ export const RegisterAccount1 = (props: RegisterAccount1Props) => {
         </Link>
       case Pages.EMAIL_REGISTRATION:
       case Pages.GOOGLE_REGISTRATION:
-        return <IconButton className='back-button' onClick={()=>setPage(Pages.CHOOSE_REGISTRATION)}>
-          <IconSvg icon='arrowBack'/>
-        </IconButton>
+        return (
```
Comment (conundrumgirl) [inline src/components/RegisterAccount1.tsx:61]: format + size change for compatibility

File: src/components/SourceApp.tsx [hunk 14]
```diff
@@ -1,10 +1,10 @@
-import { ThemeOptions } from '@mui/material'
+import { DeprecatedThemeOptions } from '@mui/material'
```
Comment (conundrumgirl) [inline src/components/SourceApp.tsx:1]: same as above re: ThemeOptions type compatibility

File: src/types.d.ts [hunk 15]
```diff
@@ -0,0 +1,28 @@
+import '@mui/material/styles';
```
Comment (conundrumgirl) [inline src/types.d.ts:1]: module augmentation to support the SRC theme types

File: src/style/theme.ts [hunk 16]
```diff
@@ -0,0 +1,19 @@
+import { createTheme, Theme } from "@mui/material";
```
Comment (conundrumgirl) [inline src/style/theme.ts:1]: this is a placeholder file for now for common control styles to be merged with particular app related styles/palette

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**